### PR TITLE
Updates for VirtualMesh ingress gateway selection

### PR DESCRIPTION
**Pull request by harveyxia**
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh-enterprise/issues/938

### CODE REVIEW

File: api/discovery/v1/mesh.proto [hunk 1]
```diff
@@ -203,8 +203,12 @@ message MeshStatus {
         // The port on the ingress gateway Destination designated for receiving cross cluster traffic.
         uint32 destination_port = 3;
 
+        // The external facing port on the ingress gateway Destination designated for receiving cross cluster traffic.
+        // May differ from the destination_port if the Kubernetes Service is of type NodePort.
+        uint32 external_destination_port = 4;
+
         // The port on the ingress gateway's backing Workload(s) designated for receiving cross cluster traffic.
-        uint32 container_port = 4;
```
Comment (EItanya) [inline api/discovery/v1/mesh.proto:207]: nit: shouldn't break number

Comment (harveyxia) [reply api/discovery/v1/mesh.proto:207]: discussed offline, this is fine given that it's a break between beta versions

**Comment by harveyxia** [comment]
Follow up GME PR here, https://github.com/solo-io/gloo-mesh-enterprise/pull/939

File: pkg/mesh-networking/translation/istio/mesh/federation/federation_translator.go [hunk 2]
```diff
@@ -127,7 +127,7 @@ func (t *translator) buildGateway(
 		Spec: networkingv1alpha3spec.Gateway{
 			Servers: []*networkingv1alpha3spec.Server{{
 				Port: &networkingv1alpha3spec.Port{
-					Number:   ingressContainerPort,
+					Number:   ingressDestinationPort,
 					Protocol: defaultGatewayProtocol,
 					Name:     defaults.IstioGatewayTlsPortName,
```
Comment (EItanya) [inline pkg/mesh-networking/translation/istio/mesh/federation/federation_translator.go:132]: What if the name of this port is different, isn't it settable VIA our selector API

**Review by EItanya — APPROVED** [review]

**Pull request merged**